src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -673,13 +672,77 @@ void CodeGen::genSSEIntrinsic(GenTreeHWIntrinsic* node)
 void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
 {
     NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    GenTree*       op1         = node->gtGetOp1();
+    GenTree*       op2         = node->gtGetOp2();
+    regNumber      targetReg   = node->gtRegNum;
+    var_types      targetType  = node->TypeGet();
     var_types      baseType    = node->gtSIMDBaseType;
-    instruction    ins         = INS_invalid;
+    instruction    ins         = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+    regNumber      op1Reg      = REG_NA;
+    regNumber      op2Reg      = REG_NA;
+    emitter*       emit        = getEmitter();
+    int            ival        = -1;
 
-    genConsumeOperands(node);
+    if ((op1 != nullptr) && !op1->OperIsList())
+    {
+        op1Reg = op1->gtRegNum;
+        genConsumeOperands(node);
+    }
 
     switch (intrinsicID)
     {
+        case NI_SSE2_CompareLessThan:
+        {
+            assert(op1 != nullptr);
+            assert(op2 != nullptr);
+            op2Reg = op2->gtRegNum;
+
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    ival = Compiler::ivalOfHWIntrinsic(intrinsicID);
+                    emit->emitIns_SIMD_R_R_R_I(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg, ival);
+                    break;
+
+                case TYP_INT:
+                case TYP_UINT:
+                case TYP_LONG:
+                case TYP_ULONG:
+                case TYP_BYTE:
+                case TYP_UBYTE:
+                case TYP_SHORT:
+                case TYP_USHORT:
+                    emit->emitIns_SIMD_R_R_R(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg);
```

Comment:
Once you transform `NI_SSE2_CompareLessThan` to `NI_SSE2_CompareGreaterThan` in the importer, you do need to deal with the codegen on integers.

Reply:
Thanks, got it